File: 20221217/20221217.c
/*
---- 2022/12/17 ----
Q: Swap two numbers (e.g. a=3 and b=5) without a third variable. 
*/
#include <stdio.h>

int main()
{
    int a=3;
    int b=5;
    printf("a=%d,b=%d \n",a,b);
    a=a-b; 
    b=a+b; 
    a=(-a)+b;
    printf("a=%d,b=%d \n",a,b);
    return 0;
}

File: 20221217/Python/20221217.py
#---- 2022/12/17 ----
#Q: Swap two numbers (e.g. a=3 and b=5) without a third variable. 

def main():
    a=3
    b=5
    print(f"a={a},b={b}")
    a=a-b 
    b=a+b 
    a=(-a)+b
    print(f"a={a},b={b}")
if __name__== "__main__":
    main()